default/python/AI/AIstate.py

```diff
@@ -95,7 +101,7 @@ class AIstate(object):
     via boost. If desiring to store a reference to a UniverseObject store its
     object id instead; for enum values store their int conversion value.
     """
-    version = 3
+    version = 4
```

Comment:
It looks like a version update got missed at some point in the past, and the previous one should have been 4, this one 5.  (Note that the previous one was version 3, but line 68 was at the same time rejected version 3 as no longer supported.  If this one is left as version 4, then when trying to load a savegame from this version then I expect that lines 73-74 would cause an exception.

Reply:
The versioning should be correct but the code probably isn't as clear as it could/should be.
You seem to worry about the line
```
    if version <= 3:
        raise ConversionError("The AI savegame version is no longer supported.")
```
The variable ```version``` here is the *target* version to convert the state into (the docstring states that but the variable could be renamed for further clarity), so if the ```current_version``` of the savegame state is already 3 or higher, that line would never be executed.

I'll add revising that code to make it a little clearer to my todo list.

Reply:
Ah, as I had mentioned, I didn't have enough time to do a thorough review, but I should have held off from commenting on that snippet without looking at the fuller code.  Changing ```version``` in that function to ```target_version``` would be more clear, but that's pretty low priority, the biggest issue was my rushed review of just a code fragment.